.github/workflows/feature_deploy.yml

```diff
@@ -37,559 +37,151 @@ jobs:
+    # TODO: FN-1077 NOTE: we don't currently have a `feature` GitHub environment yet so use the `dev` one while we're not doing much.
+    # Restore the parameterised version throughout!
     environment:
       dev
-    env:
-      repository: azure-function-acbs
-      path: azure-functions/acbs-function
-      name: function-acbs
-
+    # environment:
+    #   name: ${{ needs.deployment-environment.outputs.environment }}
+    strategy:
+      matrix:
+        repository: 
+          - azure-function-number-generator
+          - azure-function-acbs
+        include:
+          - path: azure-functions/number-generator-function
+            repository: azure-function-number-generator
+            name: function-number-generator
+          - path: azure-functions/acbs-function
+            repository: azure-function-acbs
+            name: function-acbs
+            
     steps:
-    - name: 1. Latest commit hash
-      uses: actions/checkout@v3
-
-    - name: 2. Build and push
-      uses: docker/build-push-action@v1
-      with:
-        registry: ${{ env.acr }}
-        username: ${{ env.acr_username }}
-        password: ${{ env.acr_password }}
-        repository: ${{ env.repository }}
-        path: ${{ env.path }}
-        tags: latest
-        build_args: GITHUB_SHA=${{ github.sha }}
-        add_git_labels: true
-
-    - name: 3. ACR login
-      uses: azure/docker-login@v1
-      with:
-        login-server: ${{ env.acr }}
-        username: ${{ env.acr_username }}
-        password: ${{ env.acr_password }}
-
-    - name: 4. Tag images
-      run: |
-        from=${{ env.from }}
-        to=${{ env.environment }}
-        docker pull ${{ env.acr }}/${{ env.repository }}:$from
-        docker tag ${{ env.acr }}/${{ env.repository }}:$from ${{ env.acr }}/${{ env.repository }}:$to
-        docker push ${{ env.acr }}/${{ env.repository }}:$to
-
-    - name: 5. Azure login
-      uses: azure/login@v1
-      with:
-        creds: ${{ env.credentials }}
-
-    - name: 6. Slot
-      uses: azure/cli@v1.0.0
-      with:
-        inlineScript: |
-          az config get defaults
-          az functionapp deployment slot create --name tfs-${{ env.environment }}-${{ env.name }} --resource-group ${{ env.resource_group }} --slot ${{ github.sha }} --configuration-source tfs-${{ env.environment }}-${{ env.name }}
-          az functionapp deployment slot swap --name tfs-${{ env.environment }}-${{ env.name }} --resource-group ${{ env.resource_group }} --slot ${{ github.sha }} --action swap
-          az functionapp deployment slot delete --name tfs-${{ env.environment }}-${{ env.name }} --resource-group ${{ env.resource_group }} --slot ${{ github.sha }}
-
-    - name: 7. Restart
-      uses: azure/cli@v1.0.0
-      with:
-        inlineScript: |
-          az functionapp restart --name tfs-${{ env.environment }}-${{ env.name }} --resource-group ${{ env.resource_group }}
\ No newline at end of file
+      - name: 1. Latest commit hash
```

Comment:
This is weird.
It seems the original version didn't have the steps indented. I wonder how it worked?